ecosystem/daos/jsondao.py

```diff
@@ -93,11 +97,27 @@ def update_stars(self, url: str, tier: str, stars: int) -> List[int]:
         table = self.database.table(tier)
         return table.update({"stars": stars}, Query().url == url)
 
+    def update_labels(self, labels: List[str]) -> List[int]:
```

Comment:
instead of opening files manually you can follow the same interface with TinyDB.

Create something like that
```python
self.lables_db = TinyDB(f"{self.path}/labels.json")
```

and then use it to insert new entries if they are not there

```python

for label in labels:
     if not self.lables_db.contains(Query().name == label):
          self.labels_db.insert({"name": label, "description": ""})
```

Reply:
If I interpret https://stackoverflow.com/a/47757362 correctly, that means that TinyDB demands `"_default"` which I think it introduce noise in a file that it should be extended by hand. The `update_label` code encapsulates the mess so I would prefer to be messy in the code than in the data.